php/class-widget-posts.php

```diff
@@ -0,0 +1,683 @@
+<?php
+
+namespace CustomizeWidgetsPlus;
+
+/**
+ * Store widgets in posts as opposed to options.
+ *
+ * This makes non_autoloaded_widget_options obsolete.
+ *
+ * @package CustomizeWidgetsPlus
+ */
+class Widget_Posts {
+
+	const MODULE_SLUG = 'widget_posts';
+
+	const INSTANCE_POST_TYPE = 'widget_instance';
+
+	const ENABLED_FLAG_OPTION_NAME = 'enabled_widget_posts';
+
+	/**
+	 * @var Plugin
+	 */
+	public $plugin;
+
+	/**
+	 * @var \WP_Customize_Manager
+	 */
+	public $manager;
+
+	/**
+	 * @var \WP_Widget[]
+	 */
+	public $widget_objs;
+
+	/**
+	 * @return array
+	 */
+	static function default_config() {
+		return array();
+	}
+
+	/**
+	 * Return the config entry for the supplied key, or all configs if not supplied.
+	 *
+	 * @param string $key
+	 * @return array|mixed
+	 */
+	function config( $key = null ) {
+		if ( is_null( $key ) ) {
+			return $this->plugin->config[ static::MODULE_SLUG ];
+		} else if ( isset( $this->plugin->config[ static::MODULE_SLUG ][ $key ] ) ) {
+			return $this->plugin->config[ static::MODULE_SLUG ][ $key ];
+		} else {
+			return null;
+		}
+	}
+
+	/**
+	 * @param Plugin $plugin
+	 */
+	function __construct( Plugin $plugin ) {
+		$this->plugin = $plugin;
+
+		add_option( self::ENABLED_FLAG_OPTION_NAME, false, '', 'yes' );
+		add_action( 'widgets_init', array( $this, 'store_widget_objects' ), 90 );
```

Comment:
Since it is an object, too, it will be stored by reference anyway. So there will be almost zero space required to store this additional array, since all of the objects are strewn throughout `$wp_registered_widgets` and friends anyway.